packages/extension-ui/src/Popup/Signing/Details.tsx

```diff
@@ -31,9 +30,7 @@ function renderMethod (data: string, meta?: Metadata | null) {
     );
   }
 
-  Method.injectMethods(fromMetadata(meta));
-
-  const method = new Method(data);
+  const method = new Method(data, { meta });
```

Comment:
Is this correct? meta is not the method metadata, it is everything.

Reply:
yes, we made it so that the constructor can either accept the function metadata or the whole metadata (from which it extracts the relevant function metadata), so that it's more practical for end-users (developers)

https://github.com/polkadot-js/api/pull/1047#discussion_r297348952

Reply:
Cool, that is nice. Caters for this specific use-case.